packages/providers/src/provider.ts

```diff
@@ -206,8 +206,22 @@ export default class Provider {
     /** GraphQL endpoint of the Fuel node */
     public url: string
   ) {
+    this.operations = this.createOperations(url);
+  }
+
+  /**
+   * Create GraphQL client and set operations
+   */
+  private createOperations(url: string) {
     const gqlClient = new GraphQLClient(url);
-    this.operations = getOperationsSdk(gqlClient);
+    return getOperationsSdk(gqlClient);
+  }
+
+  /**
+   * Connect provider to a different Fuel node url
+   */
+  connect(url: string) {
+    this.operations = this.createOperations(url);
```

Comment:
Connect methods are usually _async_, validate the URL, try to _connect_ somewhere, and throw an error in case of failure. This case doesn't seem to be actually _connecting_, so I'd suggest either:

 1. Change the method name _(`setProviderUrl`?)_
 2. Make the method `async` and run a try/catch validation flow, i.e.:

```suggestion
  async connect(url: string) {
    this.operations = this.createOperations(url);
    try {
      await provider.getNodeInfo(); // or something better
    } catch (error) {
       // handle error, re-throw if necessary
    }
```

Reply:
I think this is not valid, as the creation of the provider will not be handle by developers making than to add a try catch around the creation of the wallet.

Reply:
Connect is not an async method, as it didn't and should not request data. The application layer is responsible for handling the async operations and errors on bad URL etc.